plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorSmokeTest.java

```diff
@@ -160,6 +160,51 @@ public void testDeleteRowsConcurrently()
         }
     }
 
+    private long getLatestSnapshotId(String tableName)
+    {
+        return (long) computeScalar(format("SELECT snapshot_id FROM \"%s$snapshots\" ORDER BY committed_at DESC FETCH FIRST 1 ROW WITH TIES", tableName));
+    }
+
+    @Test
+    public void testCreateOrReplaceTable()
+    {
+        String tableName = "test_create_or_replace_" + randomNameSuffix();
+        assertUpdate("CREATE TABLE " + tableName + " AS SELECT BIGINT '42' a, DOUBLE '-38.5' b", 1);
+        assertThat(query("SELECT CAST(a AS bigint), b FROM " + tableName))
+                .matches("VALUES (BIGINT '42', -385e-1)");
+
+        long v1SnapshotId = getLatestSnapshotId(tableName);
+
+        assertUpdate("CREATE OR REPLACE TABLE " + tableName + " AS SELECT BIGINT '-42' a, DOUBLE '38.5' b", 1);
+        assertThat(query("SELECT CAST(a AS bigint), b FROM " + tableName))
+                .matches("VALUES (BIGINT '-42', 385e-1)");
+
```

Comment:
It would probably be beneficial to showcase here through a "$history" query that we're still dealing with the same table.